datafusion/src/datasource/object_store/local.rs

```diff
@@ -82,12 +82,15 @@ impl ObjectReader for LocalFileReader {
         &self,
         start: u64,
         length: usize,
-    ) -> Result<Box<dyn Read + Send + Sync>> {
+    ) -> Result<Box<dyn BufRead + Send + Sync>> {
         // A new file descriptor is opened for each chunk reader.
         // This okay because chunks are usually fairly large.
         let mut file = File::open(&self.file.path)?;
         file.seek(SeekFrom::Start(start))?;
-        Ok(Box::new(file.take(length as u64)))
+
+        let file = BufReader::new(file.take(length as u64));
```

Comment:
It seems like this is the actual fix, right? Is the change to require the `BufRead` trait needed?

Reply:
Not sure. Rerunning some benchmarks now without the trait.

Reply:
Yes - looks like `BufRead` wasn't needed :tada: